### ISSUE TITLE
NullCells not created in case of NULL values when exporting from MySQL to SIARDDK

### ISSUE THREAD

**Issue by andreaskring** [keeps/db-preservation-toolkit#208]
Description: When exporting a MySQL database containing only the table
```
mysql> select * from table1;
+----+------+
| id | str  |
+----+------+
|  0 | NULL |
+----+------+
1 row in set (0.00 sec)
```
to SIARDDK an error occurs (NullPointerException in this [line](https://github.com/magenta-aps/db-preservation-toolkit/blob/nullCellProblem/dbptk-core/src/main/java/com/databasepreservation/modules/siard/out/content/SIARDDKContentExportStrategy.java#L270) since `simpleCell.getSimpleData()` returns `null`) since the NULL value in the table above is not stored in a NullCell, but in a SimpleCell. In SIARDDK, NULL cells cannot be omitted from the table[n].xml files as in SIARD 2.0, but should be included like this:
```
<?xml version="1.0" encoding="utf-8"?>
<table xsi:schemaLocation=...>
  <row>
    <c1>0</c1>
    <c2 xsi:nil="true"/>
  </row>
</table>
```
The problem could be handled by checking for null values in the JDBCImportModule in [this](https://github.com/keeps/db-preservation-toolkit/blob/master/dbptk-core/src/main/java/com/databasepreservation/modules/jdbc/in/JDBCImportModule.java#L1243) line as suggested [here](https://github.com/magenta-aps/db-preservation-toolkit/commit/933817f1939cbac85fe001b3fb512eb3b0071c31). A similar approach is used in the JDBCImportModule in approximately line 1220 for `SimpleTypeBoolean`. The checking for null values should probably also occur other places in the `convertRawToCell` method in the JDBCImportModule.

@chalkos In you oppinion - should the NullCell checking as described above be implemented?

**Comment by chalkos** [comment]
Yes. it should.

I'm not working on dbptk at the moment. feel free to change it yourself.

Suggestions (untested):

In line https://github.com/keeps/db-preservation-toolkit/blob/master/dbptk-core/src/main/java/com/databasepreservation/modules/jdbc/in/JDBCImportModule.java#L1221 change `cell = new SimpleCell(id, null);` to `cell = new NullCell(id);`

and in line https://github.com/keeps/db-preservation-toolkit/blob/master/dbptk-core/src/main/java/com/databasepreservation/modules/jdbc/in/JDBCImportModule.java#L1243 add
```
if (rawData.wasNull()) {
  cell = new NullCell(id);
}
```

That should fix the problem for SimpleTypeBoolean and SimpleTypeString coming from JDBCImportModule.

**Comment by andreaskring** [comment]
@chalkos Ok - thanks. I will fix it

**Comment by PhillipTommerholt** [comment]
@chalkos @andreaskring 

This is also a pilot blocker.

**Comment by andreaskring** [comment]
This have beeb fixed in pull request #211
